flytekit/types/structured/structured_dataset.py

```diff
@@ -25,6 +25,77 @@
 DF = typing.TypeVar("DF")  # Dataframe type
 
 
+"""
+
+# Python 3.7/3.8
+from typing_extensions import Annotated
+
+# Python 3.9+
+from typing import Annotated
```

Comment:
nit: we're already "paying" the price of bringing in the `typing_extensions` module (regardless of the python version), so we can default to always importing `Annotated` from `typing_extensions`.

Reply:
no this isn't us.  this is user code.  we have to make sure that whatever users do, we can find out the proper annotations